apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx

```diff
@@ -111,6 +127,51 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
     setAdvancedSettingsOpen(!advancedSettingsOpen);
   };
 
+  const recordSectionSetupEvent = (e, section) => {
+    e.preventDefault();
+    const initial = initialSection.current;
+    // We do not currently store version year on the section, and the version dropdown
+    // does not update it. We will need to query all course offerings or set up a new
+    // course offerings controller function in order to populate these values.
+    let versionYear;
+    let initialVersionYear;
+    if (isNewSection) {
+      analyticsReporter.sendEvent(EVENTS.COMPLETED_EVENT, {
+        sectionUnitId: section.course?.unitId,
+        sectionCurriculumLocalizedName: section.course?.displayName,
+        sectionCurriculum: section.course?.courseOfferingId, //this is course Offering id
+        sectionCurriculumVersionYear: section.course?.versionYear,
+        sectionGrade: section.grade ? section.grade[0] : null,
+        sectionLockSelection: section.restrictSection,
+        sectionName: section.name,
+        sectionPairProgramSelection: section.pairingAllowed,
+      });
+    }
+    if (
+      (section.course?.courseOfferingId &&
+        initial &&
+        section.course?.courseOfferingId !==
+          initial.course?.courseOfferingId) ||
+      (section.course?.unitId &&
+        initial &&
+        section.course?.unitId !== initial.course?.unitId)
```

Comment:
this might be a good place to add comments, maybe in a followup when you add the tests. this logic is hard for me to follow

Reply:
I totally agree that it's worth a bit of explanation- I added a few lines that *hopefully* communicate what's happening!